src/borg/testsuite/archiver.py

```diff
@@ -3731,9 +3731,88 @@ def do_asserts(output, can_compare_ids):
+                assert not any(get_changes('input/link_target_removed', joutput))
+
+            # The inode has two links and the file contents changed. Borg
+            # should notice the changes in both links. However, the symlink
+            # pointing to the file is not changed.
+            assert any(('type', 'modified') in chg.items() for chg in get_changes('input/empty', joutput))
+            if are_hardlinks_supported():
+                assert any(('type', 'modified') in chg.items() for chg in get_changes('input/hardlink_contents_changed', joutput))
+            if are_symlinks_supported():
+                assert not any(get_changes('input/link_target_contents_changed', joutput))
+
+            # Added a new file and a hard link to it. Both links to the same
+            # inode should appear as separate files.
+            assert any(('type', 'added') in chg.items() and ('size', 2048) in chg.items() for chg in get_changes('input/file_added', joutput))
+            if are_hardlinks_supported():
+                # assert any(('added', 2048) in chg.items() for chg in get_changes('input/hardlink_added', joutput))
+                assert any(('type', 'added') in chg.items() and ('size', 2048) in chg.items() for chg in get_changes('input/hardlink_added', joutput))
+
+            # check if a diff between non-existent and empty new file is found
+            assert any(('type', 'added') in chg.items() and ('size', 0) in chg.items() for chg in get_changes('input/file_empty_added', joutput))
+
+            # The inode has two links and both of them are deleted. They should
+            # appear as two deleted files.
+            assert any(('type', 'removed') in chg.items() and ('size', 256) in chg.items() for chg in get_changes('input/file_removed', joutput))
+            if are_hardlinks_supported():
+                assert any(('type', 'removed') in chg.items() and ('size', 256) in chg.items() for chg in get_changes('input/hardlink_removed', joutput))
+
+            # Another link (marked previously as the source in borg) to the
+            # same inode was removed. This should not change this link at all.
+            if are_hardlinks_supported():
+                assert not any(get_changes('input/hardlink_target_removed', joutput))
+
+            # Another link (marked previously as the source in borg) to the
+            # same inode was replaced with a new regular file. This should not
+            # change this link at all.
+            if are_hardlinks_supported():
+                assert not any(get_changes('input/hardlink_target_replaced', joutput))
+
         do_asserts(self.cmd('diff', self.repository_location + '::test0', 'test1a'), True)
         # We expect exit_code=1 due to the chunker params warning
         do_asserts(self.cmd('diff', self.repository_location + '::test0', 'test1b', exit_code=1), False)
+        # test '--json-lines' output mode
```

Comment:
you can leave away the comment, it does not add information that is not already obvious from next line.